edge-apps/iframe/static/js/main.js

```diff
@@ -0,0 +1,20 @@
+/* global  screenly */
+
+function extractUrlFromIframeSetting (settingValue) {
+  if (settingValue.includes('<iframe')) {
+    const match = settingValue.match(/<iframe.*?src=["'](.*?)["']/)
+    if (match && match[1]) {
+      return match[1]
+    }
+  } else {
+    return settingValue
+  }
+  return null
+}
+document.addEventListener('DOMContentLoaded', function () {
+  const iframeSettingValue = screenly.settings.iframe || 'iframe URL not set'
+  const iframeUrl = extractUrlFromIframeSetting(iframeSettingValue)
```

Comment:
In case of 'iframe URL not set', iframeUrl will be 'iframe URL not set'

and line no 18 will try to set the text as the url

Reply:
in` 18:iframeElement.src = iframeUrl` which will assign the fetched **iframeUrl** to **iframeElement.src** so that It can be used in the main iframe tag we have in the index.

Reply:
Yeah, Now I got your point and yes it will set the URL as text.

Reply:
Do you recommend any other method? Thanks